sympy/polys/matrices/tests/test_rref.py

```diff
@@ -0,0 +1,193 @@
+from sympy import ZZ, QQ
+from sympy.polys.matrices import DM, DomainMatrix
+from sympy.polys.matrices.dense import ddm_irref_den, ddm_irref
+from sympy.polys.matrices.sdm import sdm_irref
+
+import pytest
+
+
+RREF_EXAMPLES = [
+    (
+        'zz_1',
+         DM([[1, 2, 3]], ZZ),
+         DM([[1, 2, 3]], ZZ),
+         ZZ(1),
+    ),
+
+    (
+        'zz_2',
+         DomainMatrix([], (0, 0), ZZ),
+         DomainMatrix([], (0, 0), ZZ),
+         ZZ(1),
+    ),
+
+    (
+        'zz_3',
+        DM([[1, 2],
+            [3, 4]], ZZ),
+        DM([[-2,  0],
+            [ 0, -2]], ZZ),
+        ZZ(-2),
+    ),
+
+    (
+        'zz_4',
+        DM([[1, 2, 3],
+            [4, 5, 6],
+            [7, 8, 9]], ZZ),
+        DM([[-3,  0,  3],
+            [ 0, -3, -6],
+            [ 0,  0,  0]], ZZ),
+        ZZ(-3),
+    ),
+
+    (
+        'zz_5',
+        DM([[0, 0, 0],
+            [0, 0, 0],
+            [1, 0, 0]], ZZ),
+        DM([[1, 0, 0],
+            [0, 0, 0],
+            [0, 0, 0]], ZZ),
+        ZZ(1),
+    ),
+
+    (
+        'zz_6',
+        DM([[0, 0, 0],
+            [0, 0, 0],
+            [0, 0, 0]], ZZ),
+        DM([[0, 0, 0],
+            [0, 0, 0],
+            [0, 0, 0]], ZZ),
+        ZZ(1),
+    ),
+
+    (
+        'zz_7',
+        DM([[1, 1, 0],
+            [0, 0, 2],
+            [0, 0, 0]], ZZ),
+        DM([[2, 2, 0],
+            [0, 0, 2],
+            [0, 0, 0]], ZZ),
+        ZZ(2),
+    ),
+
+    (
+        'zz_7',
+        DM([[2, 2, 0],
+            [0, 0, 2],
+            [0, 0, 0]], ZZ),
+        DM([[4, 4, 0],
+            [0, 0, 4],
+            [0, 0, 0]], ZZ),
+        ZZ(4),
+    ),
+
+    (
+        'zz_7',
+        DM([[2, 2, 0],
+            [0, 2, 2],
+            [0, 0, 2]], ZZ),
+        DM([[8, 0, 0],
+            [0, 8, 0],
+            [0, 0, 8]], ZZ),
+        ZZ(8),
+    ),
+
```

Comment:
Might be good to have more non-square examples, which can sometimes catch certain issues.
E.g.
```python
    (
        'zz_10',
        DM([[1, 2, 3],
            [4, 5, 6],
            [7, 8, 9],
            [10, 11, 12]], ZZ),
        DM([[-3, 0, 3],
            [0, -3, -6],
            [0, 0, 0],
            [0, 0, 0]], ZZ),
        ZZ(-3),
    ),

    (
        'zz_11',
        DM([[1, 2, 3],
            [4, 5, 6],
            [7, 8, 9],
            [10, 11, 13]], ZZ),
        DM([[-3, 0, 0],
            [0, -3, 0],
            [0, 0, -3],
            [0, 0, 0]], ZZ),
        ZZ(-3),
    ),

    (
        'zz_12',
        DM([[1, 2, 4, 3],
            [4, 5, 10, 6],
            [7, 8, 16, 9]], ZZ),
        DM([[-3, 0, 0, 3],
            [0, -3, -6, -6],
            [0, 0, 0, 0]], ZZ),
        ZZ(-3),
    ),

    (
        'zz_13',
        DM([[1, 2, 4, 3],
            [4, 5, 10, 6],
            [7, 8, 17, 9]], ZZ),
        DM([[-3, 0, 0, 3],
            [0, -3, 0, -6],
            [0, 0, -3, 0]], ZZ),
        ZZ(-3),
    ),
```

Reply:
Thanks. I've added these.